### ISSUE TITLE
Issue with justify-content: space-between

### ISSUE THREAD

**Issue by heruan** [vaadin/vaadin-ordered-layout#66]
If the `spacing` attribute is set, setting `justify-content: space-between` doesn't work as expected since the `:host([theme~="spacing"])::before` pseudo-element is considered as content and justified itself.

Seems like the Spac~~e~~ing Odyssey isn't over yet, @jouni 🙁

**Comment by heruan** [comment]
This and also #52 makes me confident that using `display: grid` would be a better fit for layouts. I guess there are compatilibity concerns about it? Seems like it's "just" IE having issues with that, but why not using polyfills then?

**Comment by jouni** [comment]
😬 ugh, yikes. Yeah, that’s an unfortunate limitation as well.

Though, I’m wondering about the use case again – I assume you want the layout to expand but the components inside not, just be spaced out equally. But when the layout shrinks, you want to make sure there’s some minimum space between them?

CSS grid would be quite perfect for this, but yeah, IE support has been the reason to avoid it. Would be nice to prototype a solution based on it, though, and see how much any needed polyfills would cause performance issues.

**Comment by heruan** [comment]
Exactly, I need an equal space between but also be sure when the viewport is smaller (e.g. mobile) the default spacing kicks-in. Of course, it's easy to workaround… using `display: grid` 🙂 

I'll try to prototype something, since I'm basically using that instead of horizontal/vertical layouts on new projects and it's *really* game changing (mainly for responsiveness).

**Comment by rolandoisidoro** [comment]
@web-padawan , sorry for opening a duplicate issue as I wasn't thorough enough on my search. I limited it to the Vaadin component point-of-view and didn't check it from the CSS property.

I see now that this is a long standing issue. Is there any workaround for this?

On the other hand, issue #67 mentions a possible solution, but was kept on hold since January 2019 due to "Safari 9 support". Is this limitation still in place?

**Comment by web-padawan** [comment]
We no longer have a requirement to support Safari 9 in the next LTS so I hope we will be able to use CSS grid.

**Comment by rolfsmeds** [comment]
Actually I think it would be more straightforward to refactor layouts to use the `gap` property coming to flexbox. Last I checked, refactoring them to use css grid has some problems of its own that are not related to browser support (although I can't remember what those were right now). Dropping IE11 and old Edge support in next LTS clears the main obstacles for refactoring, and I'd expect all supported browsers to support flexbox gap by the time the next LTS goes into alpha.

**Comment by rolandoisidoro** [comment]
@web-padawan and @rolfsmeds , both of you mention "next LTS". Any solution for v14?

**Comment by rolfsmeds** [comment]
You could override the css used by horizontal-layout and vertical-layout to use the css grid approach in #67, _or_, as a smaller tweak, override the spacing attribute implementation to use the gap property instead of applying margins for child elements, if your browser support requirements allow it.
